enumerables.rb

```diff
@@ -0,0 +1,132 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_each
+    x = 0
+    return to_enum unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    while x < list.length
+      yield(list[x])
+      x += 1
+    end
+    list
+  end
+
+  def my_each_with_index
+    x = 0
+    return to_enum unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    while x < list.length
+      yield(list[x], x)
+      x += 1
+    end
+    list
+  end
+
+  def my_select
+    result = []
+    return to_enum unless block_given?
+
+    my_each do |x|
+      result << x if yield(x)
+    end
+    result
+  end
+
+  def my_all?(input = nil)
+    my_each do |x|
+      return false if block_given? && !yield(x)
+
+      if !block_given? && input.nil?
+        return false unless x
+      elsif input
+        return false unless input_check(x, input)
+      end
+    end
+    true
+  end
+
+  def my_any?(input = nil)
+    my_each do |x|
+      return true if block_given? && yield(x)
+
+      if !block_given? && input.nil?
+        return true if x
+      elsif !block_given? && input
+        return true if input_check(x, input)
+      end
+    end
+    false
+  end
+
+  def my_none?(input = nil)
+    my_each do |x|
+      return false if block_given? && yield(x)
+
+      if !block_given? && input.nil?
+        return false if x
+      elsif !block_given? && input
+        return false if input_check(x, input)
+      end
+    end
+    true
+  end
```

Comment:
[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
def my_none?(arg = nil, &block)
!my_any?(arg, &block)
end
```

To make this work we have to use block.call(element) instead of yield(element) in my_any?